src/components/KFileUpload/KFileUpload.vue

```diff
@@ -235,6 +252,35 @@ const resetInput = (): void => {
 
   emit('file-removed')
 }
+
+const setLabelAttributes = () => {
+  /**
+   * Temporary fix for the issue where we can't use v-bind-once to pass id to a custom element (KInput)
+   * TODO: remove this once useId is released in Vue 3.5
+   */
+  if (!attrs.id) {
+    const inputElementId = fileInputElement.value?.$el?.querySelector('input')?.id
+
+    if (inputElementId) {
+      labelElement.value?.$el?.setAttribute('for', inputElementId)
+    }
+  }
+}
+
+watch(fileInputKey, async () => {
+  await nextTick()
+  setLabelAttributes()
+})
+
+watch(() => attrs.id, async () => {
+  fileInputKey.value++
+  await nextTick()
+  setLabelAttributes()
```

Comment:
issue: you're calling `setLabelAttributes` in both functions, so you're essentially doing 2 nextTick awaits and 2 setLabelAttributes calls every single time?

Reply:
Good catch, fixed